### ISSUE TITLE
Exports always uses UTC as timezone instead of the selected Report Timezone

### ISSUE THREAD

**Issue by vibear7463** [metabase/metabase#13677]
**Describe the bug**
My team is using metabase version 0.37.2 and got an issue when downloading query result with date_trunc function inside it.
We only met this issue when using PostgreSQL, I tried on Bigquerry but its still alright.

**To Reproduce**
- First, we write query like this:
`select current_date, date_trunc('day',current_date) as day, date_trunc('week',current_date) as week, date_trunc('month',current_date) as month, date_trunc('quarter',current_date) as quarter
`
- The result on metabase seems fine

- Then we tried to export this result to excel or csv, it got an issue
![image](https://user-images.githubusercontent.com/73939202/98103811-26e89000-1ec8-11eb-9bcb-8032d9f9a183.png)

_Plus, Our timezone is Asia, Bangkok, in case you need to know._

**Expected behavior**
The result's timestamp exported to csv/xlsx should be 12:00:00 instead of 17:00:00.

**Screenshots**
![image](https://user-images.githubusercontent.com/73939202/98103783-1f28eb80-1ec8-11eb-8b25-1c8b4eb290e2.png)

**Information about your Metabase Installation:**

- Databases: PostgreSQL
- Metabase version: 0.37.2

**Severity**
Its blocking our automation for reporting so please fix this issue as soon as possible

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by flamber** [comment]
Hi @vibear7463 
Please post "Diagnostic Info" from Admin > Troubleshooting.
The frontend might figure out this is a date, so it displays it like that, but the backend doesn't always know unless you cast it:
`date_trunc('day',current_date)::date`
Or am I completely misunderstanding the problem?

**Comment by vibear7463** [comment]
At previous version we did not meet this issue, let me check with our team
about the diagnostic info

**Comment by flamber** [comment]
@vibear7463 Then please write which version you were previously using.

**Comment by thucnc** [comment]
I would say this bug is a inconsistency of showing date value: Metabase shows values in correct configured timezone while the export file shows values in UTC timezone. 
The version 0.36.6 works fine.

**Comment by flamber** [comment]
It only worked before 0.36.8 if the question was saved. Otherwise it would cause inconsistency and other problems.
Since 0.36.8 the PR #13447 fixed a bunch of export issues, so it is likely a regression of that.
I'm not sure if the is specific to Postgres - I have a feeling that it might not be.

**Comment by thucnc** [comment]
Hi @flamber, this bug affects many our automation flows, so may we have its fix in the next release ? 
Sorry if this request bothers you, we rely on metabase a lot.
Thanks.

**Comment by flamber** [comment]
@thucnc You need to revert to your backup. Metabase does not officially support downgrading, but you should be able to downgrade. Remember to backup before upgrading/downgrading or doing any other major changes.

**Comment by caochangshan** [comment]
The version 0.36.6 works fine.
When will it be fixed?

**Comment by flamber** [comment]
@caochangshan As you can see right above your comment, this has been scheduled for 0.37.3

**Comment by caochangshan** [comment]
The download result report invalid time zone setting version has not been fixed in 0.37.3

**Comment by flamber** [comment]
@caochangshan When the issue is closed, then it will be fixed and will have a milestone attached.

**Comment by camsaul** [comment]
Fixed by #13986

**Comment by rlotun** [comment]
@caochangshan apologies for the milestone scope change, but this bug is fixed and will appear in 37.4 which should be released sometime next week.

**Comment by vibear7463** [comment]
Hello, we did update our metabase to version 37.4 but still meet this issue
when using API
Please check it again, thank you.

**Comment by flamber** [comment]
Reopening. I'm not sure if this is specific only to Postgres+Redshift, but it's definitely been taking a while to figure out what's going on and that if you run everything in UTC, then everything looks correct.

Here's an example from our internal instance:
```
select current_date, date_trunc('day',current_date) as day, date_trunc('week',current_date) as week, date_trunc('month',current_date) as month, date_trunc('quarter',current_date) as quarter, current_setting('TIMEZONE') AS "tz_db"
```
It will display something like this:
![image](https://user-images.githubusercontent.com/1447303/105769489-2a873280-5f5e-11eb-9fd5-cd00a3d5fac6.png)

But exporting as CSV/XLSX on Postgres will show like this:
```
2021-01-25,2021-01-25T08:00:00Z,2021-01-25T08:00:00Z,2021-01-01T08:00:00Z,2021-01-01T08:00:00Z,US/Pacific
```
While exporting as CSV on Redshift is correct, but exporting as XLSX mess up the timezone like with Postgres.

**Comment by atanas0** [comment]
@flamber Thanks for reopening this. It really is the issue that we've been experiencing.
Do you need info about our installation and steps-to-reproduce?

**Comment by atanas0** [comment]
Databases: Postgres
Metabase version: 0.37.7

**Comment by atanas0** [comment]
Hi @flamber Am I understanding correctly that the team will discuss this issue on Monday?

**Comment by flamber** [comment]
@atanas0 Please downgrade. I don't know when a fix will be implemented or when it will be part of a release.

**Comment by caochangshan** [comment]
The download result report invalid time zone setting version has not been fixed in 0.37.4

**Comment by flamber** [comment]
@caochangshan Please post "Diagnostic Info" from Admin > Troubleshooting.

**Comment by atanas0** [comment]
Worked around it by converting date to text.

**Comment by thucnc** [comment]
@flamber do you have any milestone for fixing this bug? This bug is critical for us to upgrade metabase. Thanks

**Comment by flamber** [comment]
@thucnc Currently no, but it has high priority and will likely be part of one of the patch releases of 0.38.x

**Comment by thucnc** [comment]
> @thucnc Currently no, but it has high priority and will likely be part of one of the patch releases of 0.38.x

So may you add this issue into milestone 0.38.1 , @flamber ?

**Comment by akshay1007** [comment]
My version is v0.38.0.1 but same issue I am facing any update on this @flamber

**Comment by flamber** [comment]
When the issue is closed with a milestone, then it will be fixed. I don't have a timeline for this.

**Comment by model2020** [comment]
@thucnc  ru

**Comment by caochangshan** [comment]
Too slow, for months

**Comment by vibear7463** [comment]
still nothing update guys ?

**Comment by camsaul** [comment]
@flamber are you sure this is happening on XLSX with Postgres? I've spent hours pulling out my hair using your exact query trying to reproduce it.

![image](https://user-images.githubusercontent.com/1455846/113069865-1934ef00-9176-11eb-8e2c-7600a049b5e1.png)

**Comment by camsaul** [comment]
Fixed by #15415
